lib/libzpool/kernel.c

```diff
@@ -291,70 +186,43 @@ kstat_set_raw_ops(kstat_t *ksp,
 void
 mutex_init(kmutex_t *mp, char *name, int type, void *cookie)
 {
-	ASSERT3S(type, ==, MUTEX_DEFAULT);
-	ASSERT3P(cookie, ==, NULL);
-	mp->m_owner = MTX_INIT;
-	mp->m_magic = MTX_MAGIC;
-	VERIFY3S(pthread_mutex_init(&mp->m_lock, NULL), ==, 0);
+	VERIFY0(pthread_mutex_init(&mp->m_lock, NULL));
+	mp->m_owner = 0;
```

Comment:
Yeah, this definitely not portable but I couldn't find a pthreads function or way to set an invalid tid.  `bzero()` a good idea, the should work everywhere.